packages/kbn-scout/src/playwright/page_objects/maps_page.ts

```diff
@@ -9,10 +9,21 @@
 
 import { ScoutPage } from '..';
 
+const DEFAULT_MAP_LOADING_TIMEOUT = 10_000;
+
 export class MapsPage {
   constructor(private readonly page: ScoutPage) {}
 
   async gotoNewMap() {
-    await this.page.gotoApp('maps/map');
+    return this.page.gotoApp('maps/map');
+  }
+
+  async waitForRenderComplete() {
+    // first wait for the top level container to be present
+    await this.page.locator('div#maps-plugin').waitFor({ timeout: DEFAULT_MAP_LOADING_TIMEOUT });
+    // then wait for the map to be fully rendered
+    return this.page
+      .locator('div[data-dom-id][data-render-complete="true"]')
```

Comment:
Are we sure this is always only matching the map? Wouldn't it be safer to check for this only under the maps top level container?

Reply:
I was also not sure in the beginning, but this code in maps container is the only place in the Kibana repo:

https://github.com/elastic/kibana/blob/0c69f75a3795e69aa7084911eeacc5b4cc0166f2/x-pack/platform/plugins/shared/maps/public/connected_components/map_container/map_container.tsx#L205-L206

I believe it is only matching the map container, but it is not tied to Maps page (can be an embeddable in other app?)

Reply:
Alright, I guess it's alright then :+1: